synapseclient/client.py

```diff
@@ -1031,6 +1011,20 @@ def store(self, obj, **kwargs):
                 # But becomes an update() due to conflict
                 properties['dataFileHandleId'] = bundle['entity']['dataFileHandleId']
 
+            #update the file_handle metadata if the FileEntity's FileHandle id has changed
+            if '_file_handle' in local_state and properties['dataFileHandleId'] != local_state['_file_handle'].get('id', None):
+                local_state['_file_handle'] = self._getFileHandle(properties['dataFileHandleId'])
+                #check if we alredy have the filehandleid cached somewhere
+                cached_path = self.cache.get(properties['dataFileHandleId'])
+                if cached_path is None:
```

Comment:
if _uploadFileHandle() is called add the file to the cache. this logic is currently only in store() and get()